File: src/main/java/at/hrechny/predictionsbot/util/NameUtils.java
package at.hrechny.predictionsbot.util;

import java.util.regex.Pattern;
import org.apache.commons.lang3.RandomStringUtils;

public class NameUtils {

  private NameUtils() {
  }

  private static final String ALLOWED_SYMBOLS = "[\\d -/:-~À-ž\\u0370-\\u03FFА-я∂]";
  private static final Pattern NAME_PATTERN = Pattern.compile("^" + ALLOWED_SYMBOLS + "{3,20}$");

  public static boolean isValid(String name) {
    return name != null && NAME_PATTERN.matcher(name).matches();
  }

  public static String formatName(String name) {
    var nameBuilder = new StringBuilder();
    if (name != null) {
      for(var c  : name.toCharArray()) {
        if (String.valueOf(c).matches(ALLOWED_SYMBOLS)) {
          nameBuilder.append(c);
        }
      }
    }

    var formattedName = nameBuilder.toString().trim();
    if (isValid(formattedName)) {
      return formattedName;
    } else {
      return RandomStringUtils.randomAlphanumeric(6);
    }
  }
}

File: src/main/kotlin/at/hrechny/predictionsbot/util/NameUtils.kt
package at.hrechny.predictionsbot.util

import java.util.regex.Pattern
import org.apache.commons.lang3.RandomStringUtils

object NameUtils {
    private const val ALLOWED_SYMBOLS = "[^\\t\\n]"
    private val namePattern: Pattern = Pattern.compile("^$ALLOWED_SYMBOLS{2,20}$")

    @JvmStatic
    fun isValid(name: String?): Boolean = name != null && namePattern.matcher(name).matches()

    @JvmStatic
    fun formatName(name: String?): String {
        val nameBuilder = StringBuilder()
        if (name != null) {
            for (char in name.toCharArray()) {
                if (char.toString().matches(Regex(ALLOWED_SYMBOLS))) {
                    nameBuilder.append(char)
                }
            }
        }

        val formattedName = nameBuilder.toString().trim()
        return if (isValid(formattedName)) {
            formattedName
        } else {
            RandomStringUtils.secure().nextAlphanumeric(6)
        }
    }
}
